src/components/spreadsheet/spreadsheet.ts

```diff
@@ -253,12 +252,14 @@ export class Spreadsheet extends Component<SpreadsheetProps, SpreadsheetChildEnv
 
   private bindModelEvents() {
     this.model.on("update", this, () => this.render(true));
-    this.model.on("notify-ui", this, this.onNotifyUI);
+    this.model.on("notify-ui", this, this.env.notifyUser);
+    this.model.on("raise-error-ui", this, this.onRaiseErrorUI);
```

Comment:
you can inline the handler
```suggestion
    this.model.on("raise-error-ui", this, this.env.raiseError);
```